doc/source/docs/howto_build_docs.rst

```diff
@@ -4,105 +4,116 @@
-------------
+   Since large parts of the main documentation are obtained from NumPy via
+   ``import numpy`` and examining the docstrings, you will need to first
+   :ref:`build <building-from-source>`, and install it so that the correct version is imported.
+   NumPy has to be re-built and re-installed every time you fetch the latest version of the
+   repository, before generating the documentation. This ensures that the NumPy version and
+   the git repository version are in sync.
+
+   Note that you can e.g. install NumPy to a temporary location and set
+   the PYTHONPATH environment variable appropriately.
+   Alternatively, if using Python virtual environments (via e.g. ``conda``,
+   ``virtualenv`` or the ``venv`` module), installing NumPy into a
+   new virtual environment is recommended.
 
-If you obtained NumPy via git, get also the git submodules that contain
-additional parts required for building the documentation::
+2) Dependencies
 
-    git submodule update --init
+   All of the necessary dependencies for building the NumPy docs can be installed
+   with::
 
-In addition, building the documentation requires the Sphinx extension
-`plot_directive`, which is shipped with Matplotlib_. This Sphinx extension can
-be installed by installing Matplotlib. You will also need Python>=3.6.
+       pip install -r doc_requirements.txt
 
-Since large parts of the main documentation are obtained from numpy via
-``import numpy`` and examining the docstrings, you will need to first build
-NumPy, and install it so that the correct version is imported.
+   We currently use Sphinx_ for generating the API and reference
+   documentation for NumPy. In addition, building the documentation requires
+   the Sphinx extension `plot_directive`, which is shipped with Matplotlib_. SciPy_
+   is installed since some parts of the documentation require SciPy functions.
 
-After NumPy is installed, install SciPy since some of the plots in the random
-module require `scipy.special` to display properly.
+3) Submodules
 
-Note that you can eg. install NumPy to a temporary location and set
-the PYTHONPATH environment variable appropriately.
-Alternatively, if using Python virtual environments (via e.g. ``conda``,
-``virtualenv`` or the ``venv`` module), installing numpy into a
-new virtual environment is recommended.
-All of the necessary dependencies for building the NumPy docs can be installed
-with::
+   If you obtained NumPy via git, also get the git submodules that contain
+   additional parts required for building the documentation::
 
-    pip install -r doc_requirements.txt
+       git submodule update --init
+
+   NumPy's documentation uses several custom extensions to Sphinx.  These
+   are shipped in the ``sphinxext/`` directory (as git submodules), and are automatically
+   enabled when building NumPy's documentation.
+
+   If you want to make use of these extensions in third-party
+   projects, they are available on PyPi_ as the numpydoc_ package.
```

Comment:
I had asked about this in the issue #19104 and @melissawm had directed me to the mathjax issue #19036. From the discussion there I figured that the submodule may come back in the future and so we have to keep the section in the doc? I must confess I didn't follow the entire discussion so I was not sure whether to keep the section or not.

Reply:
The lines above this are correct --- NumPy does use a submodule again; but these lines are specific to older submodules that are no longer used.

Reply:
Thanks for the clarification!

Reply:
The second sentence (about getting numpydoc on pypi) is also no longer relevant and should be removed - numpy itself now gets numpydoc from pypi rather than including it as a submodule. It looks like you can eliminate the PyPI and numpydoc links as well.

Reply:
Will it be worth it to add a line explaining that the submodule is for `mathjax`, which helps us render mathematical notation? We could also add a line in the dependencies section saying that we use the `numpydoc` Sphinx extension to render the doctrings.

Reply:
> Will it be worth it to add a line explaining that the submodule is for mathjax

I would just as soon avoid this, so we avoid this whole problem again in the future if we decide to remove the mathjax submodule :)

> We could also add a line in the dependencies section saying that we use the numpydoc Sphinx extension to render the doctrings.

:+1: